docs/intro/tutorial05.txt

```diff
@@ -512,6 +515,7 @@ class:
             """
             create_question(question_text="Future question.", days=30)
             response = self.client.get(reverse('polls:index'))
+            self.assertEqual(response.status_code, 200)
```

Comment:
`assertContains()` already checks the status code.

Reply:
Oh okay, I was not aware of this (I am new to Django). So what do you think of adding this information to the tutorial so that future readers don’t think it was an oversight like me?

Reply:
Tutorial is not the right place for all details, it's a step of steps to complete a small project, see https://diataxis.fr/tutorials/.

Reply:
> `assertContains()` already checks the status code.

So why does the tutorial check the status code [here](https://docs.djangoproject.com/en/4.0/intro/tutorial05/#testing-our-new-view)?

```python
class QuestionIndexViewTests(TestCase):
    def test_no_questions(self):
        """
        If no questions exist, an appropriate message is displayed.
        """
        response = self.client.get(reverse('polls:index'))
        self.assertEqual(response.status_code, 200)
        self.assertContains(response, "No polls are available.")
        self.assertQuerysetEqual(response.context['latest_question_list'], [])
```

Reply:
To show that `status_code` is available in the `response`. Please see [TicketClosingReasons/UseSupportChannels](https://code.djangoproject.com/wiki/TicketClosingReasons/UseSupportChannels) for ways to get help with Django usage, PR is not the right place to ask usage/support questions.

Reply:
Well the tutorial [already informs the reader](https://docs.djangoproject.com/en/4.0/intro/tutorial05/#the-django-test-client) that the `status_code` attribute is available in responses:

```
>>> # get a response from '/'
>>> response = client.get('/')
Not Found: /
>>> # we should expect a 404 from that address; if you instead see an
>>> # "Invalid HTTP_HOST header" error and a 400 response, you probably
>>> # omitted the setup_test_environment() call described earlier.
>>> response.status_code
404
>>> # on the other hand we should expect to find something at '/polls/'
>>> # we'll use 'reverse()' rather than a hardcoded URL
>>> from django.urls import reverse
>>> response = client.get(reverse('polls:index'))
>>> response.status_code
200
>>> response.content
b'\n    <ul>\n    \n        <li><a href="/polls/1/">What&#x27;s up?</a></li>\n    \n    </ul>\n\n'
>>> response.context['latest_question_list']
<QuerySet [<Question: What's up?>]>
```

So checking it prior to the first `assertContains` [here](https://docs.djangoproject.com/en/4.0/intro/tutorial05/#testing-our-new-view) is misleading because it is redundant:


```
class QuestionIndexViewTests(TestCase):
    def test_no_questions(self):
        """
        If no questions exist, an appropriate message is displayed.
        """
        response = self.client.get(reverse('polls:index'))
        self.assertEqual(response.status_code, 200)
        self.assertContains(response, "No polls are available.")
        self.assertQuerysetEqual(response.context['latest_question_list'], [])
```

And then in the same class *not* checking it before the next `assertContains` is confusing because the reader thinks it is an oversight of the tutorial (that is what leads me to opening this PR in the first place so this is not an hypothetical situation):

```
def test_future_question(self):
        """
        Questions with a pub_date in the future aren't displayed on
        the index page.
        """
        create_question(question_text="Future question.", days=30)
        response = self.client.get(reverse('polls:index'))
        self.assertContains(response, "No polls are available.")
        self.assertQuerysetEqual(response.context['latest_question_list'], [])
```

And finally in the same class *not* checking when there is no `assertContains` at all is even more confusing because it *is* an oversight of the tutorial:

```
    def test_past_question(self):
        """
        Questions with a pub_date in the past are displayed on the
        index page.
        """
        question = create_question(question_text="Past question.", days=-30)
        response = self.client.get(reverse('polls:index'))
        self.assertQuerysetEqual(
            response.context['latest_question_list'],
            [question],
        )
```